src/ui/common/src/components/tables/PaginatedSearchTable.tsx

```diff
@@ -125,53 +153,196 @@ export const PaginatedSearchTable: React.FC<PaginatedSearchTableProps> = ({
     return value;
   };
 
-  if (searchQuery.length > 0) {
-    filteredRows = data.data.filter((rowItem) => {
-      return shouldInclude(rowItem, searchQuery, searchColumn);
+  React.useEffect(() => {
+    if (searchQuery.length > 0) {
+      const filteredRows = data.data.filter((rowItem) => {
+        return shouldInclude(rowItem, searchQuery, searchColumn);
+      });
+
+      setRows(filteredRows);
+    } else {
+      setRows(data.data);
+    }
+  }, [searchQuery]);
+
+  React.useEffect(() => {
+    if (
+      !sortConfig.sortColumn ||
+      !sortConfig.sortColumn.name ||
+      !sortConfig.sortType
+    ) {
+      setRows(data.data);
+      return;
+    }
+
+    const sortedRows = [...rows].sort((r1, r2) => {
+      const col = sortConfig.sortColumn;
+      let v1: PaginatedSearchTableRow | PaginatedSearchTableElement = r1;
```

Comment:
What do v1 and v2 stand for?

Reply:
value1 and value 2. I can update to be clearer.